core/templates/services/audio-player.service.ts

```diff
@@ -16,162 +16,235 @@
+      } else {
+        this._currentTrack.seek(0);
       }
-    };
-
-    var _rewind = function(seconds) {
-      if (_currentTrack) {
-        var currentSeconds = _currentTrack.progress * _currentTrack.duration;
-        if (currentSeconds - seconds > 0) {
-          var rewindedProgress = (
-            (currentSeconds - seconds) / _currentTrack.duration);
-          _currentTrack.progress = rewindedProgress;
-        } else {
-          _currentTrack.progress = 0;
-        }
+    }
+  }
+
+  private _forward(seconds: number): void {
+    if (this._currentTrack) {
+      const currentSeconds = this._currentTrack.seek();
+      if (typeof currentSeconds !== 'number') {
+        return;
       }
-    };
-
-    var _forward = function(seconds) {
-      if (_currentTrack) {
-        var currentSeconds = _currentTrack.progress * _currentTrack.duration;
-        if (currentSeconds + seconds < _currentTrack.duration) {
-          var forwardedProgress = (
-            (currentSeconds + seconds) / _currentTrack.duration);
-          _currentTrack.progress = forwardedProgress;
-        }
+      if (currentSeconds + seconds < this._currentTrack.duration()) {
+        this._currentTrack.seek(currentSeconds + seconds);
       }
-    };
+    }
+  }
 
-    return {
-      load: function(filename) {
-        return $q(function(resolve, reject) {
-          _load(filename, resolve, reject);
-        });
-      },
-      play: function() {
-        _play();
-      },
-      pause: function() {
-        _pause();
-      },
-      stop: function() {
-        _stop();
-      },
-      rewind: function(seconds) {
-        _rewind(seconds);
-      },
-      forward: function(seconds) {
-        _forward(seconds);
-      },
-      getCurrentTime: function() {
-        if (_currentTrack) {
-          return Math.round(_currentTrack.currentTime);
-        } else {
-          return 0;
-        }
-      },
-      getAudioDuration: function() {
-        if (_currentTrack && _currentTrack.audio) {
-          return Math.round(_currentTrack.audio.duration);
-        } else {
-          return 0;
-        }
-      },
-      getProgress: function() {
-        if (!_currentTrack) {
-          return 0;
-        }
-        return _currentTrack.progress;
-      },
-      setProgress: function(progress) {
-        if (_currentTrack) {
-          _currentTrack.progress = progress;
-        }
-      },
-      isPlaying: function() {
-        return Boolean(_currentTrack && !_currentTrack.paused);
-      },
-      isTrackLoaded: function() {
-        return Boolean(_currentTrack);
-      },
-      clear: function() {
-        _currentTrack = null;
-        _currentTrackFilename = null;
-      },
-      get onAutoplayAudio() {
-        return _autoplayAudioEventEmitter;
+  load(filename: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this._load(filename, resolve, reject);
+    });
+  }
+
+  play(): void {
+    this._play();
+  }
```

Comment:
Can we just have one public function?(Earlier we had to do this as the play function was used in multiple places so cannot define it in the return it)

[Ditto for other places]